### DISCUSSION TITLE
Compilation failure with workshop code

### DISCUSSION THREAD

**Discussion by JasperTF1** [PicnicSupermarket/error-prone-support#1237] [Q&A] [answered]
For the workshop today, I ran `mvn clean install` on the workshop branch. I got the following compilation failure:

    Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
    /C:/Users/jaspe/Documents/Techforce1/error-prone-support/refaster-compiler/src/main/java/tech/picnic/errorprone/refaster/plugin/RefasterRuleCompilerTaskListener.java:[142,27] cannot find symbol
      symbol:   method lastIndexOf(byte)
      location: variable flatName of type com.sun.tools.javac.util.Name

Is it known what I need to do to get past this compilation failure?

**Comment by Stephan202** [comment]
Thanks for asking @JasperTF1! My first hunch is that this may be due to a Java incompatibility. Can you please share the output of `mvn -version`? If we can reproduce the issue, hopefully a fix is easily found :crossed_fingers:

**Reply by JasperTF1** [reply, marked as answer]
Thanks, I'm at the workshop now and have been helped by @rickie in person. The problem was indeed that I was using Java 22 (on windows). Java 21 also didn't work, but Java 17 does.

**Reply by Stephan202** [reply]
Closing the loop on this: not sure why JDK 21 failed, but I just opened a PR for JDK 22 compatibility: #1277.
